File: 6-kyu/delete-occurrences-of-an-element-if-it-occurs-more-than-n-times/c++/solution.cpp
#include <map>

std::vector<int> deleteNth(std::vector<int> arr, int n)
{
    std::map<int, int> value_counts;
    std::vector<int> ret;
    for (auto it = arr.begin(); it != arr.end(); it++) {
        if (value_counts[*it] < n) {
            ret.push_back(*it);
            value_counts[*it]++;
        }
    }
    return ret;
}

File: 6-kyu/delete-occurrences-of-an-element-if-it-occurs-more-than-n-times/python/solution.py
def delete_nth(order, max_e):
    counts = {}
    ret = []
    for n in order:
        if n not in counts:
            counts[n] = 1
            ret.append(n)
        else:
            if counts[n] < max_e:
                ret.append(n)
                counts[n] += 1
    return ret